### ISSUE TITLE
Get not only message but also cause

### ISSUE THREAD

**Issue by dlsrb6342** [Hakky54/log-captor#4]
Hi. 
I wanna get throwable class from logger.
I debugged and found throwable class name from `listAppender.list.get(0).getThrowableProxy().getClassName()`.
It would be nice that `LogCaptor#getLogs()` returns a model that have message and throwable class name.
What do you think?

**Comment by Hakky54** [comment]
Hi,

This sounds like a great idea and as you already mentioned these specific messages and class names are already available within the listappender. Currently `LogCaptor#getLogs()` returns a `List` of `Strings` and by returning a model it will have a breaking change for other end-users. I don't prefer that actually... Another option would be introducing additional methods, but that will maybe break the simplicity of the api...

I have analysed the logs when an exception occurs with basic logging, see below for an example:
```
09:31:34.540 [main] ERROR nl.altindag.log.service.exception.ZooService - Caught unexpected exception
java.io.IOException: KABOOM!
	at nl.altindag.log.service.exception.ZooService.tryToSpeak(ZooService.java:23)
	at nl.altindag.log.service.exception.ZooService.sayHello(ZooService.java:16)
	at nl.altindag.log.LogCaptorShould.captureLoggingEventsContainingException(LogCaptorShould.java:75)
        // and a lot more lines from the stacktrace
```

I was wondering what you think of the solution of concatenating the log message return by `LogCaptor#getLogs()` into the following lines:
```
Caught unexpected exception
java.io.IOException: KABOOM!
```

Will that be useful for you?

See here for the detailed implementation with example unit test: https://github.com/Hakky54/log-captor/pull/5

**Comment by dlsrb6342** [comment]
I think It is mixing 2 different concerns into one. 
I prefer adding new method like `List<CapturedLogEvent> getCapturedLogEvents()`.
```java
class CapturedLogEvent {
    String message;
    Level level;
    Throwable throwable;
}
```

And, your implementation can be a solution for me :)

**Comment by Hakky54** [comment]
That's a nice suggestion as an alternative indeed, I like it more than the current implementation as it is less magic and more explicit. The throwable contains already the specific exception class type and also the message, so it will fit your needs. I don't prefer to have a Level as an Object type within the CapturedLogEvent as it will expose the internal API (Logback) to the library users. I think it would be better to have it as a String value. I made some changes, please let me know what you think of it and if this solution is better for your use case.

**Comment by dlsrb6342** [comment]
I reviewed your changes. and looks fine for me :)

**Comment by Hakky54** [comment]
Glad to hear that this solution is what you are looking for. I have made a release, so this feature will be available from version 2.2.0 onwards. Good luck asserting your log messages!

**Comment by dlsrb6342** [comment]
Awesome! 🚀 
Thanks :)
